src/GeneralStateTestsFiller/stEWASMTests/createFromTransactionExceedGasLimitFiller.yml

```diff
@@ -0,0 +1,52 @@
+# Try creating a contract, the creation cost is greater than gas limit
+---
+createFromTransactionExceedGasLimit:
+  env:
+    currentCoinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    currentDifficulty: '0x020000'
+    currentGasLimit: '89128960'
+    currentNumber: '1'
+    currentTimestamp: '1000'
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+  pre:
+    # tx sender
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
+      code: ''
+      nonce: ''
+      storage: {}
+  expect:
+    - indexes:
+        data: !!int -1
+        gas: !!int -1
+        value: !!int -1
+      network:
+        - ALL
+      result:
+        a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+          balance: '99995025681'
+        # should not create contract, exceeds max code size
+        6295ee1b4f6dd65047762f924ecd367c17eabf8f:
+          shouldnotexist: "1"
+  transaction:
+    data:
+    - |
+      (module
+        (import "ethereum" "return" (func $return (param i32 i32)))
+        (memory 100)
+        (export "memory" (memory 0))
+        (export "main" (func $main))
+        (func $main
+          (call $return
+            (i32.const 0)
+            (i32.const 24576))
+        )
+      )
+    gasLimit:
+    - '0x4be6ef'
```

Comment:
Could you include slightly more on the gas math here? What's the gas cost for this tx and how is it calculated?

Reply:
After I wrote the first createFromTransaction test case @axic suggested more scenarios that we could tests, Returning Zero bytes, returning two bytes and random bytes.

Reply:
@lrettig I fixed the comments, also added some more to show the contract creation is exceeding the gas limit. Actually I had trouble calculating the gas usage in this test case, I couldn't calculate why it needs 4974320 (0x4be6f0) gas to create the contract, but the test shows that if the gas limit is less than that (4974319 / 0x4be6ef), the creation fails.